### ISSUE TITLE
Lidarr: Unable to update to v0.8.0.2041

### ISSUE THREAD

**Issue by houndtt** [SynoCommunity/spksrc#4409]
_For new Package Requests, see the guidelines_  

### Setup
_Package Name:_ Lidarr
_Package Version:_ 20200921-5 (v0.7.2.1878)

_NAS Model:_ DS916+
_NAS Architecture:_ INTEL Pentium N3710
_DSM version:_ 6.2.3-25426 Update 3

### Expected behavior
From the System menu, choose Updates. Once there click on the Install Latest button which would upgrade from v0.7.2.1878 to v0.8.0.2041.

### Actual behavior
Update downloads and extracts successfully. A backup is created and the update attempted, however the actual update does not take place. Lidarr says it is restarting to complete the update but based on the uptime of the app, the restart is not completed. A manual restart is successful but the update is not installed.

### Steps to reproduce
_1._ From the System menu, choose Updates
_2._ Click on the Install Latest button
_3._ Wait on the process to complete but it does not (even after several hours)

#### Package log
_Check Package Center or `/usr/local/{package}/var/`_
```
Tue Jan 26 08:42:57 -04 2021
Starting lidarr command env PATH=/var/packages/mono/target/bin:/volume1/@appstore/lidarr/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/syno/sbin:/usr/syno/bin:/usr/local/sbin:/usr/local/bin HOME=/volume1/@appstore/lidarr/var LD_LIBRARY_PATH=/volume1/@appstore/lidarr/lib /var/packages/mono/target/bin/mono /volume1/@appstore/lidarr/share/Lidarr/Lidarr.exe 
[Info] Bootstrap: Starting Lidarr - /volume1/@appstore/lidarr/share/Lidarr/Lidarr.exe - Version 0.7.2.1878 
[Info] Router: Application mode: Interactive 
[Info] MigrationLogger: *** Migrating data source=/volume1/@appstore/lidarr/var/.config/Lidarr/lidarr.db;cache size=-10000;datetimekind=Utc;journal mode=Wal;pooling=True;version=3 *** 
[Info] MigrationLogger: *** Migrating data source=/volume1/@appstore/lidarr/var/.config/Lidarr/logs.db;cache size=-10000;datetimekind=Utc;journal mode=Wal;pooling=True;version=3 *** 
[Info] OwinHostController: Listening on the following URLs: 
[Info] OwinHostController:   http://*:8686/ 
[Info] LidarrBootstrapper: Starting Web Server 
[Warn] MonoDebugCheck: Mono is not running with --debug switch 
[Info] InstallUpdateService: Deleting old update files 
[Info] InstallUpdateService: Downloading update 0.8.0.2041 
[Info] InstallUpdateService: Verifying update package 
[Info] InstallUpdateService: Update package verified successfully 
[Info] InstallUpdateService: Extracting Update package 
[Info] InstallUpdateService: Update package extracted successfully 
[Info] BackupService: Starting Backup 
[Info] InstallUpdateService: Preparing client 
[Info] InstallUpdateService: Starting update client /tmp/lidarr_update/Lidarr.Update.exe 
[Info] InstallUpdateService: Lidarr will restart shortly. 
[Info] InstallUpdateService: Updater Arguments: 31364 /tmp/lidarr_update /volume1/@appstore/lidarr/share/Lidarr/Lidarr.exe  
[Info] RssSyncService: Starting RSS Sync 
[Info] DownloadDecisionMaker: Processing 200 releases 
[Info] RssSyncService: RSS Sync Completed. Reports found: 200, Reports grabbed: 0 
[Info] InstallUpdateService: Deleting old update files 
[Info] InstallUpdateService: Downloading update 0.8.0.2041 
[Info] InstallUpdateService: Verifying update package 
[Info] InstallUpdateService: Update package verified successfully 
[Info] InstallUpdateService: Extracting Update package 
[Info] InstallUpdateService: Update package extracted successfully 
[Info] BackupService: Starting Backup 
[Info] InstallUpdateService: Preparing client 
[Info] InstallUpdateService: Starting update client /tmp/lidarr_update/Lidarr.Update.exe 
[Info] InstallUpdateService: Lidarr will restart shortly. 
[Info] InstallUpdateService: Updater Arguments: 31364 /tmp/lidarr_update /volume1/@appstore/lidarr/share/Lidarr/Lidarr.exe  
[Info] LifecycleService: Restart requested. 
[Info] Bootstrap: Starting Lidarr - /volume1/@appstore/lidarr/share/Lidarr/Lidarr.exe - Version 0.7.2.1878 
[Info] Router: Application mode: Interactive 
[Info] MigrationLogger: *** Migrating data source=/volume1/@appstore/lidarr/var/.config/Lidarr/lidarr.db;cache size=-10000;datetimekind=Utc;journal mode=Wal;pooling=True;version=3 *** 
[Info] MigrationLogger: *** Migrating data source=/volume1/@appstore/lidarr/var/.config/Lidarr/logs.db;cache size=-10000;datetimekind=Utc;journal mode=Wal;pooling=True;version=3 *** 
[Info] OwinHostController: Listening on the following URLs: 
[Info] OwinHostController:   http://*:8686/ 
[Info] LidarrBootstrapper: Starting Web Server 
[Warn] MonoDebugCheck: Mono is not running with --debug switch 
...
[Info] HousekeepingService: Running housecleaning tasks 
[Info] Database: Vacuuming Log database 
[Info] Database: Log database compressed 
[Info] Database: Vacuuming Main database 
[Info] Database: Main database compressed 
...
[Info] InstallUpdateService: Deleting old update files 
[Info] InstallUpdateService: Downloading update 0.8.0.2041 
[Info] InstallUpdateService: Verifying update package 
[Info] InstallUpdateService: Update package verified successfully 
[Info] InstallUpdateService: Extracting Update package 
[Info] InstallUpdateService: Update package extracted successfully 
[Info] BackupService: Starting Backup 
[Info] InstallUpdateService: Preparing client 
[Info] InstallUpdateService: Starting update client /tmp/lidarr_update/Lidarr.Update.exe 
[Info] InstallUpdateService: Lidarr will restart shortly. 
[Info] InstallUpdateService: Updater Arguments: 4307 /tmp/lidarr_update /volume1/@appstore/lidarr/share/Lidarr/Lidarr.exe /nobrowser 
[Info] RssSyncService: Starting RSS Sync 
[Info] DownloadDecisionMaker: Processing 200 releases 
[Info] RssSyncService: RSS Sync Completed. Reports found: 200, Reports grabbed: 0 
```
#### Other logs
_`/var/log/synopkg.log`_
```
2021/01/26 08:01:37	stop lidarr: begin to stop version 20200921-5
2021/01/26 08:01:38	stop lidarr 20200921-5 Begin start-stop-status stop
2021/01/26 08:01:39	stop lidarr 20200921-5 End start-stop-status stop ret=[0]
2021/01/26 08:01:39	stop lidarr 20200921-5 Begin unload apparmor
2021/01/26 08:01:39	stop lidarr 20200921-5 End unload apparmor ret=[0]
2021/01/26 08:01:39	stop lidarr: stop version 20200921-5 successfully, result 0
2021/01/26 08:42:56	start lidarr: begin to start version 20200921-5
2021/01/26 08:42:57	start lidarr 20200921-5 Begin pre-load apparmor
2021/01/26 08:42:57	start lidarr 20200921-5 End pre-load apparmor ret=[0]
2021/01/26 08:42:57	start lidarr 20200921-5 Begin start-stop-status start
2021/01/26 08:43:02	start lidarr 20200921-5 End start-stop-status start ret=[0]
2021/01/26 08:43:02	start lidarr: start version 20200921-5 successfully, result 0
```

**Comment by houndtt** [comment]
For further testing I did a backup, uninstall, re-install, restore then update and the same issue has occurred. As such, I believe this to be a persistent issue. I have uploaded the update logs from a fresh install and hopefully this will help:

Lidarr.debug.txt - https://pastebin.com/Yjp2QNef
Lidarr.trace.txt - https://pastebin.com/kTNkYDDZ

**Comment by houndtt** [comment]
From the testing I conducted to build a new package at the updated version the application does not start successfully. I put the logs of the build attempt here: https://github.com/SynoCommunity/spksrc/pull/4410#issuecomment-768274430

I don't know if this suggests a problem with Lidarr itself but would appreciate any feedback on the attempt.

**Comment by houndtt** [comment]
Okay, so the bug report in the Lidarr repository confirmed this as a Lidarr bug which was already fixed in Radarr which they will pull into the code (https://github.com/lidarr/Lidarr/issues/1943#issuecomment-768668026). As soon as this is completed I believe that this issue can be closed.

**Comment by houndtt** [comment]
I have updated my pull request (https://github.com/SynoCommunity/spksrc/pull/4410) to include the published fix as a full upgrade to v0.8.0.2042. I have tested it as an upgrade and as a clean install and appears to work well.

You can use the packages from my test builds: https://github.com/houndtt/spksrc/releases/tag/lidarr-v0.8.0. There are builds for various architectures; to see which version you need:
https://github.com/SynoCommunity/spksrc/wiki/Architecture-per-Synology-model (Models that have (x64) listed can use the general x64 testing packages). You can perform a manual install in your Package Center to help test it out and provide feedback.
